### DISCUSSION TITLE
blog/comment-system-with-giscus/

### DISCUSSION THREAD

**Discussion by giscus** [jeremy-feng/giscus#6] [Announcements]
# blog/comment-system-with-giscus/

https://fengchao.pro/blog/comment-system-with-giscus/

**Comment by shtrausslearning** [comment]
Thanks for the guide :)

**Reply by jeremy-feng** [reply]
Glad to know that! :smile:

**Comment by withrjp** [comment]
通知邮件是 giscus 自身支持的么？需要在 github 上配置么？

**Reply by jeremy-feng** [reply]
按照上述步骤安装好就行，不需要额外配置。giscus 是利用 GitHub 的 discussion 功能来实现评论的。当 giscus 上有评论时，本质上是 GitHub 仓库里的 discussion 有新的内容，只要你没有关闭仓库的 Notification，GitHub 就会向你发邮件。

![image](https://github.com/jeremy-feng/giscus/assets/44312563/575e6563-4685-4285-996b-bc17776d25ca)

**Reply by withrjp** [reply]
试了下确实可以，3q

**Comment by LlSsJ** [comment]
问一下博主评论折叠通过giscus怎么实现呢

**Reply by jeremy-feng** [reply]
折叠评论是通过 GitHub discussions 的 Hide 功能实现的。你可以在存放评论的 GitHub 仓库中找到该评论，点击右上角选项中的"Hide"即可。

<img width="899" alt="image" src="https://github.com/jeremy-feng/giscus/assets/44312563/e033dac6-6240-4f45-a972-d16530b587d1">

参考：[managing-disruptive-comments](https://docs.github.com/en/communities/moderating-comments-and-conversations/managing-disruptive-comments)

**Reply by LlSsJ** [reply]
谢谢

**Comment by Alanhays** [comment]
你好请问图片是这么发送出来的是只能在GitHub discussions中发送吗

**Reply by jeremy-feng** [reply]
进入 GitHub discussions 的页面后，只需要将图片复制后粘贴即可，GitHub 会为图片生成一个链接。

除此之外，如果你已经有了某个图片的链接，只需要用 markdown 语法将其插入即可。

例如，我需要展示 https://dummyimage.com/600x200/ 这张图片，我可以使用下面的 markdown 语法插入这张图片：

```markdown
![Image title](https://dummyimage.com/600x200/)
```

![Image title](https://dummyimage.com/600x200/)

**Comment by XingJi-love** [comment]
# 测试评论

**Comment by moyuhualuo** [comment]
tks, 但是我招id找半天
https://docs.github.com/en/graphql/overview/explorer
这个网站登入你的账号输入下面代码解决
{
  repository(owner: "用户名", name: "仓库名") {
    id
    discussionCategories (first: 5) {
      nodes {
        name
        id
      }
    }
  }
}

**Reply by jeremy-feng** [reply]
谢谢补充～

你说的这个 id 应该是不需要自己去找的，giscus 可以自动生成配置代码。

**Comment by bg51717** [comment]
当我在一个页面下进行评论的时候，之前所有评论的人都会收到邮件通知。作者你有遇到这个情况吗？我希望改善这个情况。

**Reply by jeremy-feng** [reply]
邮件通知的机制应该是由 GitHub 的 Discussions 的通知机制决定的。确实如你所说，默认情况下，当有新评论出现时，之前所有评论的人都会收到邮件通知。我目前知道的一个方法是，若手动在 GitHub Discussions 页面中的 Notifications 栏目点击 Unsubscribe，则后续不会收到通知。但这个方法还是需要用户手动操作，如果有更好的方法，欢迎讨论~

![image](https://github.com/user-attachments/assets/c595adb2-e41c-4113-b77d-d8153544f01d)

**Comment by daxianz** [comment]
试试

**Comment by think-crow** [comment]
博主您好，请教一下，我是用hugo建的静态页面，引入giscus后，评论回复正常，但邮件通知只有摘要和地址链接，没有评论信息，这个应该怎么设置？

**Reply by jeremy-feng** [reply]
你好 :smiley:
“邮件通知只有摘要和地址链接，没有评论信息”，看起来这一步是 GitHub Discussions 邮件通知的设置导致的，应该不是 giscus 的问题。抱歉我没有太多经验，你可以搜搜看关于 GitHub Discussions 邮件通知的问题~

**Reply by think-crow** [reply]
好的，感谢回答。

**Reply by think-crow** [reply]
搞了半天，发现是正常的，原来不能用自己的GitHub 账号来测试这个邮件通知，再次感谢，暴漏自己的无知了，嘿嘿。

**Reply by jeremy-feng** [reply]
没事的，也谢谢你的反馈～
